File: json/orderReader.rb
require "json"

def main()
	filename = 'order2.json'
	if ! File.exist?(filename) then
		abort(filename + " not found")
	end
	f = File.open(filename)
	orderStr = f.read()
	parsed = JSON.parse(orderStr)
	total = 0
	tabletotals = Hash.new
	parsed.each do |order| 
		total += order['total']
		if tabletotals.has_key?(order['table']) then
			newvalue = "#{tabletotals[order['table']]}".to_f + order['total']
			tabletotals[order['table']] = newvalue
		else
			tabletotals[order['table']] = order['total']
		end
	end
	tabletotals.each do | table, amount |
		puts "Table %s had a total of £%0.2f" % [table, amount]
	end
	puts "The grand total is £%0.2f" % [total]
end

main()

File: json/orderReader.py
import os
import re
import json
def main () :
    filename = 'order2.json'
    with open(filename, 'r') as file1:
        orderStr = file1.read()
    orders = json.loads(orderStr)
    total = 0
    tabletotals = {}
    for order in orders:
        total = total + order['total']
        if order['table'] in tabletotals:
            tabletotals[order['table']] += order['total']
        else:
            tabletotals[order['table']] = order['total']
    for table, amount in tabletotals.items():
        print("Table {table} had a total of £{amount:0.2f} ".format(table=table, amount=amount))
    print("The grand total is £{gt:0.2f}".format(gt=total))
main()